### PR TITLE
Shared account s3 switch

### LINKED ISSUE #387: New Shared S3 Bucket & SNS/SQS Setup for NWM Data
I helped Diwakar test retrieving NWM data to Lambda from the new Shared S3 bucket. We'll have to make some minor tweaks to our IAM roles for this:

Edit: This ended up being a bigger deal than I originally thought, see PR #449 for details.

### PR DESCRIPTION
**Pull request by TylerSchrag-NOAA**
This pull request includes all of the IaC changes needed to switch over our HML and NWM pipelines to use the new NWS Shared Account S3 Bucket. This PR goes hand in hand with 2 commits to the env files in the vlab repo on I made on 5/17/2023, which has the harcoded bucket and sns arns used in this PR.

The one thing I have not manually tested is the change to the RnR NWM URL specified in the env file, but that seems like a straighforward change that should work. If it's alright with you, I'd say we proceed with deploying this to TI as soon as possible, and I can ask Chip how to confirm on the linux machine that it's pulling from the right spot.

**One big caveat we should clarify in the admin team meetings:** We've traditionally triggered the hml_receiver lambda function on s3 notifications from both the regular and backup HML s3 buckets. I've switched the regular bucket trigger over to the shared account bucket, but also left the trigger to our account's backup bucket in place. We should clarify if/what needs to change with the backup bucket with this change.

It's worth mentioning that we'll want to get this into UAT as soon as possible, in order to have it running there for a while before the scheduled switch-over on June 6.

**Changes include:**
- Addition of nws_shared_account_s3_bucket, nws_shared_account_nwm_sns, and nws_shared_account_hml_sns to env files in vlab repo.
- Read Access to nws_shared_account_s3_bucket in hml-ingest-role and viz-proc-pipeline-lambda roles.
- Changed hml_receiver ingest lambda function to trigger on nws_shared_account_hml_sns (and the backup S3 bucket, as mentioned above)
- Changed viz_initialize_pipeline lambda function to trigger on nws_shared_account_nwm_sns
- Clean-up of SNS module given these changes
- Update to hml_receiver ingest lambda function code to handle both SNS and S3 invocations.
- Update to viz_initialize_pipeline lambda function code to filter based on s3 key (many false starts now that we've only got one topic, but it's super cheap).
- Update to env variable in Viz EC2 user data to now set NWM bucket to nws_shared_account_s3_bucket

**One final caveat / ToDo:** I have not included any code in this PR to remove the old S3 buckets... which we should revisit once the switch is complete.

### CODE REVIEW

**Comment by CoreyKrewson-NOAA** [comment]
@AndersNilssonNoaa What is the backup HML in HydroVIS exactly? Does NCO push data to that bucket as well or is that like a replication or something on our end?

File: Core/SNS/main.tf [hunk 1]
```diff
@@ -311,122 +282,6 @@ resource "aws_sns_topic_subscription" "email-targets" {
...
-
-  topic {
-    topic_arn     = resource.aws_sns_topic.sns_topics["nwm_forcing_mrf"].arn
-    events        = ["s3:ObjectCreated:*"]
-    filter_prefix = "common/data/model/com/nwm/prod/"
-    filter_suffix = "medium_range.forcing.f240.conus.nc"
-  }
-}
-
 resource "aws_s3_bucket_notification" "nwm_max_values_bucket_notification" {
```
Comment (CoreyKrewson-NOAA) [inline Core/SNS/main.tf:285]: Can we just delete this now? TF might fail actually because the nwm_max_values key doesn't exist anymore in local.sns_topics

Comment (TylerSchrag-NOAA) [reply Core/SNS/main.tf]: Good catch! Done.

**Review by CoreyKrewson-NOAA — APPROVED** [review]

**Comment by AndersNilssonNoaa** [comment]
The Backup HML bucket is populated solely by a container on the HML ingest EC2 instances. That container counts the number of files in the primary bucket for a given hour, and if that number is lower than a threshold number, it then fetches an hourly file containing HML content and unpacks that into the backup bucket. That backup file originates from one of the National Snow Analysis installations.

**Comment by TylerSchrag-NOAA** [comment]
Thanks @AndersNilssonNoaa - That's great, as it sounds like we can just leave that alone and still working as part of this change. Just to check, is there any environment variable or reference to the main bucket in those containers that we need to update as part of this change?

Also, I updated the hml_receiver lambda code in the zip file in the repo... is there another repo where that needs to be updated as well?

**Comment by AndersNilssonNoaa** [comment]
Yes, it looks like there are some environment files in the owp-hml-ingester.tar.gz source tar ball in the deployment bucket that need to be updated. 

In owp-hml-ingester/hml_backup_fetcher/, there are some .env files that set a CBUCKET environment variable (this is the bucket that gets checked). The value of that would need to be changed to the shared source HML bucket.

**Pull request merged**